Assets/Plugins/HEBGraph/Scripts/HEBGraph.cs

```diff
@@ -0,0 +1,192 @@
+			{
+				branch.SetPosition(m_totalRadialOffset, this);
+			}
+			for (int i = 0; i < gapStarts.Count; i++)
+			{
+				gaps.Add(((gapStarts[i] + 0.5f) / m_totalRadialOffset * Mathf.PI * 2f, (gapStarts[i] + 1.5f) / m_totalRadialOffset * Mathf.PI * 2f));
+			}
+			m_groupBackground.SetGaps(gaps);
+
+			//Assign links and create lines
+			foreach (HEBGraphDataLink link in data.links)
+			{
+				if (link.toId < link.fromId ? !m_lines.ContainsKey((link.toId, link.fromId)) : !m_lines.ContainsKey((link.fromId, link.toId)))
+				{
+					UILineDrawer line = GameObject.Instantiate(m_linePrefab, m_lineParent).GetComponent<UILineDrawer>();
+					SetLineToLink(line, link);
+					m_lines.Add(link.toId < link.fromId ? (link.toId, link.fromId) : (link.fromId, link.toId), line);
+				}
+				m_leaves[link.fromId].AddLink(link);
+				m_leaves[link.toId].AddLink(link);
+			}
+
+			//Create modal manager
+			//Canvas canvas = GetComponentInParent<Canvas>();
+			//m_modal = Instantiate(m_modalPrefab, canvas.transform).GetComponent<ModalManager>();
+			//m_modal.Hide();
+		}
+
+		public HEBGraphLeaf[] CreateEntries(HEBGraphDataEntry[] a_entries, HEBGraphBranch a_parent)
+		{
+			HEBGraphLeaf[] result = new HEBGraphLeaf[a_entries.Length];
+			for (int i = 0; i < a_entries.Length; i++)
+			{
+				HEBGraphLeaf leaf = GameObject.Instantiate(m_entryPrefab, m_entryParent).GetComponent<HEBGraphLeaf>();
+				m_totalRadialOffset += leaf.Initialise(m_totalRadialOffset, a_entries[i], this, a_parent);
+				m_leaves.Add(a_entries[i].id, leaf);
+				result[i] = leaf;
+			}
+			return result;
+		}
+
+		public TextMeshProUGUI CreateGroupText(float a_angle, bool a_singleLine)
+		{
+			CircleText obj;
+			if (a_angle > 0f && a_angle < 180f)
+			{
+				obj = GameObject.Instantiate(m_groupTextPrefabDown, m_groupTextParent).GetComponent<CircleText>();
+				obj.m_angularOffset = a_angle - 180f;
+			}
+			else
+			{
+				obj = GameObject.Instantiate(m_groupTextPrefab, m_groupTextParent).GetComponent<CircleText>();
+				obj.m_angularOffset = a_angle;
+				if (a_singleLine)
+					obj.m_radius -= 30f;
+			}
+			obj.UpdateCurve();
+
+			return obj.GetComponent<TextMeshProUGUI>();
+		}
+
+		void SetLineToLink(UILineDrawer a_line, HEBGraphDataLink a_link)
+		{
+			HEBGraphLeaf from = m_leaves[a_link.fromId];
+			HEBGraphLeaf to = m_leaves[a_link.toId];
+
+			float[] splitPoints;
+			if (from.Parent == to.Parent)
+				splitPoints = m_bezierDrawer.Bezier2D(new List<Vector2> { from.Position, from.Parent.Position, to.Position }, m_lineSections);
+			else
+				splitPoints = m_bezierDrawer.Bezier2D(new List<Vector2> { from.Position, from.Parent.Position, new Vector2(0.5f, 0.5f), to.Parent.Position, to.Position }, m_lineSections);
+			Vector2[] points = new Vector2[m_lineSections];
+			points[0] = from.Position;
+			for (int i = 2; i < splitPoints.Length; i += 2)
+			{
+				points[i / 2] = new Vector2(splitPoints[i], splitPoints[i + 1]);
+			}
+			//a_line.color = m_defaultLineColour.GetColour();
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)